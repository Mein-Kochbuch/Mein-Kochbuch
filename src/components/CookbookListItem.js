import React from "react";
import styled from "styled-components/native";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";

export default function CookbookListItem({cookbook}) {

    const isDarkMode = useColorScheme() === 'dark';
    const imageSource = cookbook.thumbnail ? {uri: cookbook.thumbnail} : require("../../resources/platzhalter.png")

    return (
        <StyledTouchableOpacity isDarkMode={isDarkMode}>
            <StyledImage source={imageSource}/>
            <StyledText numberOfLines={3} ellipsizeMode={"clip"}>
                {cookbook.name}
            </StyledText>
        </StyledTouchableOpacity>
    )
}

const StyledImage = styled.Image`
  height: 100px;
  width: 100px;
  padding-top: 20px;
  padding-bottom: 20px;
  background-color: #68a0cf;
  border-radius: 10px;
  border-width: 1px;
  border-color: #fff;
`

const StyledText = styled.Text`
  max-width: 70%;
  margin-left: 12px;
  text-align: center;
  font-size: 30px;
`

const StyledTouchableOpacity = styled.TouchableOpacity`
  padding: 10px;
  margin: 12px;
  display: flex;
  align-items: center;
  flex-direction: row;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`
