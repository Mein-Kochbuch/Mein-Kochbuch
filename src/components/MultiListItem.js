import React from "react";
import styled from "styled-components/native/dist/styled-components.native.esm";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";

export default function MultiListItem({title}) {
    const isDarkMode = useColorScheme() === 'dark';

    return (
        <StyledTouchableOpacity isDarkMode={isDarkMode}>
            <StyledTest>
                {title}
            </StyledTest>
        </StyledTouchableOpacity>
    )
}

const StyledTest = styled.Text`
  text-align: center;
  font-size: 30px;
`

const StyledTouchableOpacity = styled.TouchableOpacity`
  padding: 10px;
  margin: 12px;
  display: flex;
  align-items: center;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`
