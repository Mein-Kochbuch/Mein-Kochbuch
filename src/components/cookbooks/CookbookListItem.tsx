import React from "react";
import styled from "styled-components/native";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";
import {Cookbook} from "../../models/Cookbook";

interface CookbookListItemProps {
    cookbook: Cookbook
}

export default function CookbookListItem({cookbook}: CookbookListItemProps) {

    const isDarkMode = useColorScheme() === 'dark';
    const imageSource = cookbook.thumbnail ? {uri: cookbook.thumbnail} : require("../../../resources/platzhalter.png")

    return (
        <StyledTouchableOpacity isDarkMode={isDarkMode}>
            <StyledImage source={imageSource}/>
            <StyledText numberOfLines={2} ellipsizeMode={"middle"}>
                {cookbook.name}
            </StyledText>
        </StyledTouchableOpacity>
    )
}

const StyledImage = styled.Image`
  height: 100px;
  width: 100px;
  padding-top: 20px;
  padding-bottom: 20px;
  background-color: #68a0cf;
  border-radius: 10px;
  border-width: 1px;
  border-color: #fff;
`

const StyledText = styled.Text`
  max-width: 70%;
  margin-left: 12px;
  text-align: center;
  font-size: 30px;
`

interface StyledTouchableOpacityProps {
    isDarkMode: boolean
}

const StyledTouchableOpacity = styled.TouchableOpacity<StyledTouchableOpacityProps>`
  padding: 12px;
  margin: 12px;
  height: 120px;
  display: flex;
  align-items: center;
  flex-direction: row;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`
