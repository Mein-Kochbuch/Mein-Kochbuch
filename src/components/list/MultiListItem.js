import React from "react";
import styled from "styled-components/native/dist/styled-components.native.esm";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";
import MultiListItemImage from "./MultiListItemImage";

export default function MultiListItem({title, icon}) {
    const isDarkMode = useColorScheme() === 'dark';

    return (
        <StyledTouchableOpacity isDarkMode={isDarkMode}>
            <MultiListItemImage icon={icon}/>
            <StyledTest>
                {title}
            </StyledTest>
        </StyledTouchableOpacity>
    )
}

const StyledTest = styled.Text`
  text-align: center;
  font-size: 30px;
  margin-left: 12px;
`

const StyledTouchableOpacity = styled.TouchableOpacity`
  padding: 10px;
  margin: 12px;
  display: flex;
  flex-direction: row;
  align-items: center;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`
