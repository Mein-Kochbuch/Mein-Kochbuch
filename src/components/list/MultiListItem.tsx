import React from 'react';
import styled from 'styled-components/native';
import {Colors} from 'react-native/Libraries/NewAppScreen';
import {Appearance} from 'react-native';
import MultiListItemImage from './MultiListItemImage';

interface MultiListItemProps {
  title: string;
  icon: string;
}

export default function MultiListItem({title, icon}: MultiListItemProps) {
  const isDarkMode = Appearance.getColorScheme() === 'dark';

  return (
    <StyledTouchableOpacity isDarkMode={isDarkMode}>
      <MultiListItemImage icon={icon} />
      <StyledTest>{title}</StyledTest>
    </StyledTouchableOpacity>
  );
}

const StyledTest = styled.Text`
  text-align: center;
  font-size: 30px;
  margin-left: 12px;
`;

interface StyledTouchableopacityProps {
  isDarkMode: boolean;
}

const StyledTouchableOpacity = styled.TouchableOpacity<StyledTouchableopacityProps>`
  padding: 10px;
  margin: 12px;
  display: flex;
  flex-direction: row;
  align-items: center;
  border-style: solid;
  border-color: ${props => (props.isDarkMode ? Colors.lighter : Colors.darker)};
  border-radius: 10px;
  border-width: 2px;
`;
