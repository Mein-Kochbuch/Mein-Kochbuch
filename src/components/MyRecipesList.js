import React from "react";
import {View} from "react-native";
import MultiListItem from "./MultiListItem";
import CookbookListItem from "./CookbookListItem";

export default function MyRecipesList({cookbooks}) {

    return (
        <View>
            <MultiListItem title={"All Recipes"} icon="all"/>
            <MultiListItem title={"Favorites"} icon="heart"/>
            <View style={{margin: 10}}/>
            {
                cookbooks.map((cookbook) => <CookbookListItem key={cookbook.pk} cookbook={cookbook}/>)
            }
        </View>
    )
}
