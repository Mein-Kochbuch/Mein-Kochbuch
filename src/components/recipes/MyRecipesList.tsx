import React from "react";
import {View} from "react-native";
import MultiListItem from "../list/MultiListItem";
import CookbookListItem from "../cookbooks/CookbookListItem";
import styled from "styled-components/native";
import {Cookbook} from "../../models/Cookbook";

interface MyRecipeListProps {
    cookbooks: Cookbook[]
}

export default function MyRecipesList({cookbooks}: MyRecipeListProps) {

    return (
        <View>
            <MultiListItem title={"All Recipes"} icon="all"/>
            <MultiListItem title={"Favorites"} icon="heart"/>
            <Spacer/>
            {
                cookbooks.map((cookbook) => <CookbookListItem key={cookbook.pk} cookbook={cookbook}/>)
            }
        </View>
    )
}

const Spacer = styled.View`
  margin: 12px;
`
