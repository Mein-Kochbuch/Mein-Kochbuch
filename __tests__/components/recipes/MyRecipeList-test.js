import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import MyRecipesList from "../../../src/components/recipes/MyRecipesList";
import MultiListItem from "../../../src/components/list/MultiListItem";
import CookbookListItem from "../../../src/components/cookbooks/CookbookListItem";

jest.mock("../../../src/components/list/MultiListItem", () => {
    return (props) => {
        return <div {...props}/>
    }
});

jest.mock("../../../src/components/cookbooks/CookbookListItem", () => {
    return (props) => {
        return <div {...props}/>
    }
});

const cookbooks = [{cookbook: "1", pk: "1"}, {cookbook: "2", pk: "2"}]

it('MainMenu Test', () => {
    const component = renderer.create(
        <MyRecipesList cookbooks={cookbooks}/>
    );

    expect(component.toJSON()).toMatchSnapshot()
    expect(component.toJSON().children.length).toBe(5)

    const testInstance = component.root

    expect(testInstance.findAllByType(MultiListItem)[0].props.title).toBe("All Recipes")

    expect(testInstance.findAllByType(MultiListItem)[1].props.title).toBe("Favorites")

    expect(testInstance.findAllByType(CookbookListItem)[0].props.cookbook).toStrictEqual({cookbook: "1", pk: "1"},)

    expect(testInstance.findAllByType(CookbookListItem)[1].props.cookbook).toStrictEqual({cookbook: "2", pk: "2"})

})
