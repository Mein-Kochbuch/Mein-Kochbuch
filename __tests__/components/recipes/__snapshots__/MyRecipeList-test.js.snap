// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRecipesList Test 1`] = `
<View>
  <div
    icon="all"
    title="All Recipes"
  />
  <div
    icon="heart"
    title="Favorites"
  />
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
        },
      ]
    }
  />
  <div
    cookbook={
      Object {
        "cookbook": "1",
        "pk": "1",
      }
    }
  />
  <div
    cookbook={
      Object {
        "cookbook": "2",
        "pk": "2",
      }
    }
  />
</View>
`;
